File: packages/ios-sdk/Sources/GateFlowKit/Network/Endpoint.swift
import Foundation

/// API endpoint definitions.
enum Endpoint {
    case config
    case assignments
    case confirmAssignment(String)
    case track
    case identity(String)

    var path: String {
        switch self {
        case .config: "/api/v1/config"
        case .assignments: "/api/v1/bucket/assignments"
        case .confirmAssignment(let id): "/api/v1/experiments/\(id)/confirm"
        case .track: "/api/v1/events"
        case .identity(let userId): "/api/v1/users/\(userId)"
        }
    }

    var method: String {
        switch self {
        case .config, .assignments: "GET"
        case .confirmAssignment, .track: "POST"
        case .identity: "PUT"
        }
    }
}

File: packages/android-sdk/gateflow/src/main/java/com/gateflow/sdk/network/Endpoint.kt
package com.gateflow.sdk.network

/** API endpoint definitions. */
sealed class Endpoint {
    object Config : Endpoint()
    object Assignments : Endpoint()
    data class ConfirmAssignment(val id: String) : Endpoint()
    data class Track(val count: Int) : Endpoint()

    val path: String
        get() = when (this) {
            Config -> "/api/v1/config"
            Assignments -> "/api/v1/bucket/assignments"
            is ConfirmAssignment -> "/api/v1/experiments/$id/confirm"
            is Track -> "/api/v1/events"
        }

    val method: String
        get() = when (this) {
            Config, Assignments -> "GET"
            is ConfirmAssignment, is Track -> "POST"
        }
}
